tests/testlib/s2n_ktls_test_utils.c

```diff
@@ -38,19 +42,17 @@ static S2N_RESULT s2n_test_ktls_update_prev_header_len(struct s2n_test_ktls_io_s
 
 ssize_t s2n_test_ktls_sendmsg_io_stuffer(void *io_context, const struct msghdr *msg)
 {
-    POSIX_ENSURE_REF(io_context);
     POSIX_ENSURE_REF(msg);
     POSIX_ENSURE_REF(msg->msg_iov);
 
-    /* Assuming msg_control is uint8_t is a simplification and will not work when we
-     * attempt to test the production s2n_ktls_send implementation. However, setting/parsing
-     * cmsg is critical code and will be added in a separate PR. */
-    uint8_t *record_type = (uint8_t *) msg->msg_control;
-    POSIX_ENSURE_REF(record_type);
     struct s2n_test_ktls_io_stuffer *io_ctx = (struct s2n_test_ktls_io_stuffer *) io_context;
     POSIX_ENSURE_REF(io_ctx);
     io_ctx->sendmsg_invoked_count++;
 
+    uint8_t record_type = 0;
+    struct msghdr msg_to_parse = *msg;
+    POSIX_GUARD_RESULT(s2n_ktls_get_control_data(&msg_to_parse, S2N_TLS_SET_RECORD_TYPE, &record_type));
```

Comment:
leave a comment explaining this

Reply:
https://github.com/aws/s2n-tls/pull/4146/commits/1b2f9486e7c6c5dbc82f2ba1a75d15cf5cb4c5df